### DISCUSSION TITLE
Quorum queue not remove member that not exist anymore on cluster.

### DISCUSSION THREAD

**Discussion by leomartinslima** [rabbitmq/rabbitmq-server#11213] [Questions] [answered]
I have a RabbitMQ cluster with three nodes, using quorum queues distributed among the three nodes. This cluster is set up to utilize an auto-scaling group. When an EC2 instance dies and another instance joins the RabbitMQ cluster, the quorum queues, where the node that died was the leader, still reference it as if it remained active.

In the queue details, we see three nodes online, but in the members section, there are four nodes listed.

The problem occurs with the quorum queues when another node, which was a pair of the node that died, also fails.

I have tried removing this reference using the following commands:

rabbitmq-queues shrink
rabbitmq-queues delete_member

I also tried running some commands from the other nodes like:

rabbitmq-queues grow rabbit@hostname all
rabbitmq-queues rebalance quorum

However, these attempts have not resolved the issue.

In conf the rabbitmq is set 

cluster_partition_handling = autoheal
cluster_formation.peer_discovery_backend = aws
cluster_formation.aws.region = ${AWSRegion}
cluster_formation.aws.use_autoscaling_group = true
cluster_formation.node_cleanup.only_log_warning = false
cluster_formation.discovery_retry_limit = 20
cluster_formation.discovery_retry_interval = 2000

RabbitMQ 3.13.1Erlang 26.2.3

If anyone can help me

**Comment by mkuratczyk** [comment, marked as answer]
I think `quorum_queue.continuous_membership_reconciliation.auto_remove = true` is what you're looking for, see https://github.com/rabbitmq/rabbitmq-server/pull/8218. Seems like we forgot to add these setting to the docs.

@SimonUnge can you submit the docs for this feature? Thanks

**Reply by leomartinslima** [reply]
For quorum queue is a good practice work with Autoscaling Group ? Or to quorum queue is bettler work with fixed instance ?

**Reply by mkuratczyk** [reply]
Static membership is definitely preferred. However, there are RabbitMQ users who rely on autoscaling groups, which is why AWS contributed improvements around support for autoscaling groups.

**Reply by leomartinslima** [reply]
I've  a question in  rabbitmq  version 3.13.2 for this scenario is necessary set "quorum_queue.continuous_membership_reconciliation.auto_remove = true" in rabbitmq.config ? or has it already been adjusted ?

**Reply by mkuratczyk** [reply]
no, it's `false` by default
https://github.com/rabbitmq/rabbitmq-server/blob/v3.13.2/deps/rabbit/src/rabbit_quorum_queue_periodic_membership_reconciliation.erl#L66-L67

**Comment by SimonUnge** [comment]
> @SimonUnge can you submit the docs for this feature? Thanks

Yes, let me add a section in the QQ chapter!

**Comment by mkelderm** [comment]
Is this issue solved? It looks like we have the same problem: a 3 node cluster in AWS with autoscaling. Terminating one instance, then we wait untill the new instance comes back with all the required software and config. The quorum queues still thinks the 'old' leader is down.
In the overview screen, the 3 nodes are up.

![image](https://github.com/user-attachments/assets/2d575ce2-a410-4046-8a99-6df745793ffe)

the nodes are up

![image](https://github.com/user-attachments/assets/45c69ef3-49c3-4fbf-bff7-1b26d74a7cb9)

Our config is

`
# /etc/rabbitmq/rabbitmq.conf

loopback_users.guest = true
listeners.tcp = none
listeners.ssl.default = 5672

ssl_options.cacertfile = /opt/certs/trusted.pem
ssl_options.certfile = /opt/certs/wildcard.pem
ssl_options.keyfile = /opt/certs/wildcard.key
ssl_options.secure_renegotiate = true
ssl_options.reuse_sessions = true
ssl_options.honor_cipher_order = true
ssl_options.verify = verify_none
ssl_options.fail_if_no_peer_cert = false

log.file.level = debug
log.connection.level = debug

default_user = rmqadmin
default_pass = secret

cluster_formation.peer_discovery_backend = aws
cluster_formation.aws.region = eu-west-1
cluster_formation.aws.use_autoscaling_group = true
cluster_formation.aws.use_private_ip = true

management.ssl.port = 15672
management.ssl.cacertfile = /opt/certs/rgn_trusted.pem
management.ssl.certfile = /opt/certs/wildcard.nl.rsg.pem
management.ssl.keyfile = /opt/certs/wildcard.nl.rsg.key
management.ssl.secure_renegotiate = true
management.ssl.reuse_sessions = true
management.ssl.honor_cipher_order = true
management.ssl.verify = verify_none
management.ssl.fail_if_no_peer_cert = false

quorum_queue.continuous_membership_reconciliation.auto_remove = true
quorum_queue.continuous_membership_reconciliation.enabled = true
quorum_queue.continuous_membership_reconciliation.interval = 90000

cluster_partition_handling = autoheal
cluster_formation.node_cleanup.only_log_warning = false
cluster_formation.discovery_retry_limit = 20
cluster_formation.discovery_retry_interval = 2000
`

**Comment by michaelklishin** [comment]
@mkelderm our team does not appreciate hijacking of existing issues.

We won't comment on "reports" that do not even mention RabbitMQ versions and provide no logs either.
